lib/rules/utils/ast-utils.js

```diff
@@ -1766,17 +1768,19 @@ module.exports = {
     },
 
     /**
-     * Determines whether the given raw string contains an octal escape sequence.
+     * Determines whether the given raw string contains an octal escape sequence
+     * or a non-octal decimal escape sequence ("\8", "\9").
      *
-     * "\1", "\2" ... "\7"
-     * "\00", "\01" ... "\09"
+     * "\1", "\2" ... "\7", "\8", "\9"
+     * "\00", "\01" ... "\07", "\08", "\09"
      *
      * "\0", when not followed by a digit, is not an octal escape sequence.
      * @param {string} rawString A string in its raw representation.
-     * @returns {boolean} `true` if the string contains at least one octal escape sequence.
+     * @returns {boolean} `true` if the string contains at least one octal escape sequence
+     * or at least one non-octal decimal escape sequence.
      */
     hasOctalEscapeSequence(rawString) {
```

Comment:
Given the discussion in https://github.com/eslint/eslint/issues/13740, this function should be probably renamed.

Some ideas:

* `hasOctalOrNonOctalDecimalEscapeSequence` (maybe too long)
* `hasOctalOrDecimalEscapeSequence`
* `hasLegacyNumericEscapeSequence` ("legacy" because of `\0` which is numeric but not legacy).

We could also split this into two functions (i.e. add a new one for `\8` and `\9` instead of changing this one), but it would mean that the rules should call two functions and thus run two regexes.

Reply:
Renamed to `hasOctalOrNonOctalDecimalEscapeSequence`.